core/src/main/scala/scalaz/Maybe.scala

```diff
@@ -390,6 +392,22 @@ sealed abstract class MaybeInstances extends MaybeInstances0 {
 
       override def filter[A](fa: Maybe[A])(f: A => Boolean): Maybe[A] =
         fa.filter(f)
+
+      override def alt[A](a1: => Maybe[A], a2: => Maybe[A]): Maybe[A] = altly2(a1, a2) {
+        case -\/(a) => a
+        case \/-(a) => a
+      }
+
+      override def altly2[Z, A1, A2](a1: => Maybe[A1], a2: => Maybe[A2])(f: A1 \/ A2 => Z): Maybe[Z] = a1 match {
```

Comment:
it might be worthwhile putting a line comment on apply2 / altly2 to say that they are implemented for performance. In fact you might want to implement `alt` directly for performance reasons, since it has a fairly trivial implementation.